numpy/core/arrayprint.py

```diff
@@ -1428,6 +1428,9 @@ def dtype_is_implied(dtype):
     # not just void types can be structured, and names are not part of the repr
     if dtype.names is not None:
         return False
+    
+    if dtype.byteorder != '=' and dtype.itemsize != 1:
```

Comment:
Tests are failing, I suspect `dtype.isnative` should work better?

Reply:
That works much better! I tried running `numpy/core/tests/test_arrayprint.py` with the changes in my last commit and now all the tests in that file pass.